alacritty/src/event.rs

```diff
@@ -214,30 +219,71 @@ impl Processor {
                 | WindowEvent::DoubleTapGesture { .. }
                 | WindowEvent::TouchpadPressure { .. }
                 | WindowEvent::RotationGesture { .. }
-                | WindowEvent::CursorEntered { .. }
+                | WindowEvent::PointerEntered { .. }
                 | WindowEvent::PinchGesture { .. }
-                | WindowEvent::AxisMotion { .. }
+                | WindowEvent::DragEntered { .. }
                 | WindowEvent::PanGesture { .. }
-                | WindowEvent::HoveredFileCancelled
+                | WindowEvent::DragMoved { .. }
+                | WindowEvent::DragLeft { .. }
                 | WindowEvent::Destroyed
                 | WindowEvent::ThemeChanged(_)
-                | WindowEvent::HoveredFile(_)
                 | WindowEvent::Moved(_)
         )
     }
 }
 
-impl ApplicationHandler<Event> for Processor {
-    fn resumed(&mut self, _event_loop: &ActiveEventLoop) {}
+impl Drop for Processor {
+    fn drop(&mut self) {
+        if self.config.debug.print_events {
+            info!("Exiting the event loop");
+        }
+
+        match self.gl_config.take().map(|config| config.display()) {
+            #[cfg(not(target_os = "macos"))]
```

Comment:
Why is the only used branch annotated to be exclusive to certain platforms? Is it so the `take()` is called on every platform? If that's not necessary, it seems like this should be an `if let` instead with an attr on the outside.